### DISCUSSION TITLE
Requirement that Recipes be serializable

### DISCUSSION THREAD

**Discussion by crankydillo** [openrewrite/rewrite#4348] [Q&A] [answered]
One of our recipes requires using an HTTP client.  This is (currently) maintained as an instance variable.  It is not serializable and our test made us aware of this via its serialization verification.  We made the client `transient`, and the test passed.  I see use of `transient` in various places in the code.  Is there some big reason Recipes need to be serializable?  I searched through the doc and found [this change](https://github.com/openrewrite/rewrite/commit/5e5d8ccf7b66e6f9a58b83e648a05b5392ecdb0b), but still can't piece together why this is a requirement.

I can just instantiate the client as part of a visitor, but I'd still like to know the reason for the serialization requirement.

**Comment by timtebeek** [comment, marked as answer]
Hi! OpenRewrite recipes can run in a growing number of environments, some of which leverage serialization of recipes. While indeed the OSS Maven or Gradle plugins do not use serialization beyond loading yaml recipes, others like the Moderne Platform and CLI do. To prevent any accidental incompatibilities we've made that check part of our testing framework, to surface issues early. I hope that helps answer your question!
